src/managers/input/InputManager.cpp

```diff
@@ -1390,15 +1390,27 @@ void CInputManager::setTouchDeviceConfigs(STouchDevice* dev) {
         if (wlr_input_device_is_libinput(PTOUCHDEV->pWlrDevice)) {
             const auto LIBINPUTDEV = (libinput_device*)wlr_libinput_get_device_handle(PTOUCHDEV->pWlrDevice);
 
-            const int  ROTATION = std::clamp(g_pConfigManager->getDeviceInt(PTOUCHDEV->name, "transform", "input:touchdevice:transform"), 0, 7);
-            if (libinput_device_config_calibration_has_matrix(LIBINPUTDEV))
-                libinput_device_config_calibration_set_matrix(LIBINPUTDEV, MATRICES[ROTATION]);
-
-            const auto OUTPUT = g_pConfigManager->getDeviceString(PTOUCHDEV->name, "output", "input:touchdevice:output");
-            if (!OUTPUT.empty() && OUTPUT != STRVAL_EMPTY)
-                PTOUCHDEV->boundOutput = OUTPUT;
+            const auto OUTPUT             = g_pConfigManager->getDeviceString(PTOUCHDEV->name, "output", "input:touchdevice:output");
+            const auto GTRANSFORMSET      = g_pConfigManager->getConfigValuePtr("input:touchdevice:transform")->set;
+            const auto DEVICEOUTPUTSET    = g_pConfigManager->deviceConfigExists(PTOUCHDEV->name, "output");
+            const auto DEVICETRANSFORMSET = g_pConfigManager->deviceConfigExists(PTOUCHDEV->name, "transform");
```

Comment:
why not use the same `->set` flag here? instead of the entire change to how `deviceConfigExists` works?

Reply:
1. There is no `getConfigValuePtr()` equivalent for device configurations
2. I think it would be useful to make `deviceConfigExists()` optionally be able to check for specific configuration keys.

Reply:
1. getDeviceConfig()->set
2. I don't think so. If a device config exists, all vars belonging to it exist

Reply:
1. I cannot find a function named `getDeviceConfig()`, there are `getConfigValueSafeDevice()` which returns the struct or not a pointer and `getConfigValuePtrSafe()` which first checks whether the key is for a device and then has to parse the device identifier first. As an alternative to changing the `deviceConfigExists()` function a  `getDeviceConfigValuePtr()` function could be extracted from `getConfigValuePtr()`.
2. Yes, but it is not clear whether a config key has been set by the user.

Reply:
Again, you can get the struct. We cannot receive ptrs to device configs as they are not static IIRC.

What is your problem? It returns the struct. The struct has a `set` property.

Reply:
My bad, can't use `getConfigValueSafeDevice()` as it isn't even public and `getConfigValuePtrSafe()` can return a pointer to a device config value. So another alternative besides creating `getDeviceConfigValuePtr()` would be making `getConfigValueSafeDevice()` public. I will add two more commits to give you the choice between all proposed ways to check if a device config value exists and then you can reset to the one you wish to use and merge this pr.

Reply:
Now there are three commits:
 1. cec75926ad24d571457d01bf9a478ada2d1c39b2 still uses the modified `deviceConfigExists()`
 2. 0fef48b71c022d3e829c138776727d99f3f0f17f extracts the function `getDeviceConfigValuePtr()` from `getConfigValuePtrSafe()`
 3. 4be9260c8dc09373a55e8476e64b7d632ed22102 makes `getConfigValueSafeDevice()` public

Reply:
final print of the MR looks good to me.

I squash everything anyways

If there isnt anything else, I am alright with merging this as-is

Reply:
Just a functionality test by @Smona